src/WebJobs.Script.WebHost/Properties/Resources.resx

```diff
@@ -117,6 +117,9 @@
   <resheader name="writer">
     <value>System.Resources.ResXResourceWriter, System.Windows.Forms, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089</value>
   </resheader>
+  <data name="ErrorTooManySecretBackups" xml:space="preserve">
+    <value>Repository has more then {0} non-decryptable secrets backups ({1}).</value>
```

Comment:
then => than

Rather than do a retry loop for this, can't we leverage the work you did in the other PR to identify the case where the secrets were generated by another host? E.g. before we decrypt check that these are actually our secrets using the "generatedBy" info?